rockyjs/app/simply-rocky.js

```diff
@@ -0,0 +1,1578 @@
+  ['uint8', 'type'],
+  ['uint16', 'index'],
+]);
+
+var ElementRemovePacket = new struct([
+  [Packet, 'packet'],
+  ['uint32', 'id'],
+]);
+
+var GPoint = new struct([
+  ['int16', 'x'],
+  ['int16', 'y'],
+]);
+
+var GSize = new struct([
+  ['int16', 'w'],
+  ['int16', 'h'],
+]);
+
+var GRect = new struct([
+  [GPoint, 'origin', PositionType],
+  [GSize, 'size', SizeType],
+]);
+
+var ElementCommonPacket = new struct([
+  [Packet, 'packet'],
+  ['uint32', 'id'],
+  [GPoint, 'position', PositionType],
+  [GSize, 'size', SizeType],
+  ['uint16', 'borderWidth', EnumerableType],
+  ['uint8', 'backgroundColor', ColorType],
+  ['uint8', 'borderColor', ColorType],
+]);
+
+var ElementRadiusPacket = new struct([
+  [Packet, 'packet'],
+  ['uint32', 'id'],
+  ['uint16', 'radius', EnumerableType],
+]);
+
+var ElementAnglePacket = new struct([
+  [Packet, 'packet'],
+  ['uint32', 'id'],
+  ['uint16', 'angle', EnumerableType],
+]);
+
+var ElementAngle2Packet = new struct([
+  [Packet, 'packet'],
+  ['uint32', 'id'],
+  ['uint16', 'angle2', EnumerableType],
+]);
+
+var ElementTextPacket = new struct([
+  [Packet, 'packet'],
+  ['uint32', 'id'],
+  ['uint8', 'updateTimeUnits', TimeUnits],
+  ['cstring', 'text', StringType],
+]);
+
+var ElementTextStylePacket = new struct([
+  [Packet, 'packet'],
+  ['uint32', 'id'],
+  ['uint8', 'color', ColorType],
+  ['uint8', 'textOverflow', TextOverflowMode],
+  ['uint8', 'textAlign', TextAlignment],
+  ['uint32', 'customFont'],
+  ['cstring', 'systemFont', StringType],
+]);
+
+var ElementImagePacket = new struct([
+  [Packet, 'packet'],
+  ['uint32', 'id'],
+  ['uint32', 'image', ImageType],
+  ['uint8', 'compositing', CompositingOp],
+]);
+
+var ElementAnimatePacket = new struct([
+  [Packet, 'packet'],
+  ['uint32', 'id'],
+  [GPoint, 'position', PositionType],
+  [GSize, 'size', SizeType],
+  ['uint32', 'duration'],
+  ['uint8', 'easing', AnimationCurve],
+]);
+
+var ElementAnimateDonePacket = new struct([
+  [Packet, 'packet'],
+  ['uint32', 'id'],
+]);
+
+var VoiceDictationStartPacket = new struct([
+  [Packet, 'packet'],
+  ['bool', 'enableConfirmation'],
+]);
+
+var VoiceDictationStopPacket = new struct([
+  [Packet, 'packet'],
+]);
+
+var VoiceDictationDataPacket = new struct([
+  [Packet, 'packet'],
+  ['int8', 'status'],
+  ['cstring', 'transcription'],
+]);
+
+var accelAxes = [
+  'x',
+  'y',
+  'z',
+];
+
+var clearFlagMap = {
+  action: (1 << 0),
+  text: (1 << 1),
+  image: (1 << 2),
+};
+
+
+/**
+ * SimplyRocky object provides the actual methods to render on the canvas.
+ *
+ * It's an implementation of an abstract interface used by all the other classes.
+ */
+
+var SimplyRocky = {};
+
+SimplyRocky.init = function() {
+
+  // Register this implementation as the one currently in use
+  simply.impl = SimplyRocky;
+
+  state = SimplyRocky.state = {};
+
+  state.timeOffset = new Date().getTimezoneOffset() * -60;
+
+  var clickStartTime, clickEndTime, longPress;
+  ButtonTypes.forEach(function(button) {
```

Comment:
this wont work for rocky.js on the device, will leave it here until we know the device api